neqo-transport/src/quic_datagrams.rs

```diff
@@ -97,11 +109,14 @@ impl QuicDatagrams {
         while let Some(dgram) = self.datagrams.pop_front() {
             let len = dgram.as_ref().len();
             if builder.remaining() > len {
```

Comment:
If you are making this change, then you need to change this line as well.

If you can't see why, consider what this code looks like if the frame type is 2 bytes when encoded.